django/db/models/indexes.py

```diff
@@ -135,3 +189,80 @@ def __eq__(self, other):
         if self.__class__ == other.__class__:
             return self.deconstruct() == other.deconstruct()
         return NotImplemented
+
+
+class IndexExpression(Func):
+    """Order and wrap expressions for CREATE INDEX statements."""
+    template = '%(expressions)s'
+    wrapper_classes = (OrderBy, Collate)
+
+    def set_wrapper_classes(self, connection=None):
+        # Some databases (e.g. MySQL) treats COLLATE as an indexed expression.
+        if connection and connection.features.collation_as_index_expression:
+            self.wrapper_classes = tuple([
+                wrapper_cls
+                for wrapper_cls in self.wrapper_classes
+                if wrapper_cls is not Collate
+            ])
```

Comment:
I think we can get rid of the list comprehension, can't we?
```suggestion
            self.wrapper_classes = tuple(
                wrapper_cls
                for wrapper_cls in self.wrapper_classes
                if wrapper_cls is not Collate
            )
```

Reply:
I know that this may be surprising but it is better performance to provide a list up front. This is probably not a big deal in this particular case.

Reply:
I think the list comprehension could be quicker. See [comment](https://github.com/adamchainz/flake8-comprehensions/issues/247#issuecomment-753459901)

Reply:
I stand corrected. Thanks. I didn't expect that
```
$ python -m timeit 'tuple(i for i in range(10000000))'
1 loop, best of 5: 620 msec per loop
$ python -m timeit 'tuple([i for i in range(10000000)])'
1 loop, best of 5: 556 msec per loop
```